nltk/util.py

```diff
@@ -378,9 +378,31 @@ def flatten(*args):
 # Ngram iteration
 ##########################################################################
 
+def pad_sequence(sequence, n, pad_left=False, pad_right=False, 
+                 left_pad_symbol=None, right_pad_symbol=None):
+    """
+    Returns a padded sequence of items.
+    
+    :param pad_left: whether the ngrams should be left-padded
+    :type pad_left: bool
+    :param pad_right: whether the ngrams should be right-padded
+    :type pad_right: bool
+    :param left_pad_symbol: the symbol to use for left padding (default is None)
+    :type left_pad_symbol: any
+    :param right_pad_symbol: the symbol to use for right padding (default is None)
+    :type right_pad_symbol: any
+    :rtype: iter(tuple)
+    """
+    sequence = iter(sequence)
+    if pad_left:
+        sequence = chain((left_pad_symbol,) * (n-1), sequence)
+    if pad_right:
+        sequence = chain(sequence, (right_pad_symbol,) * (n-1))
+    return sequence
+
 # add a flag to pad the sequence so we get peripheral ngrams?
 
-def ngrams(sequence, n, pad_left=False, pad_right=False, pad_symbol=None):
+def ngrams(sequence, n, **kwargs):
```

Comment:
Code style: My preference is against using kwargs in such cases. We should be documenting all the params anyway; doesn't hurt to have them in the code.